proposals/csharp-12.0/collection-expressions.md

```diff
@@ -314,6 +314,54 @@ If the target type is an *array*, a *span*, a type with a *[create method](#crea
     List<int> z = [];
     ```
 
+## Spreads
+[spreads]: #spreads
+
+```csharp
+bool b = true;
+int[] x = [.. []];            // x = []
+byte[] y = [.. b ? [1] : []]; // y = [(byte)1]
+```
+
+A *spread element* `..E` is applicable as an iterator of type `T` if any of the following hold:
+
+* There is an implicit conversion from the [*iteration type*](https://github.com/dotnet/csharpstandard/blob/standard-v6/standard/statements.md#1295-the-foreach-statement) of `E` to `T`.
+* `E` is a *collection expression* and for each element `Eᵢ` there is an implicit conversion to `T`.
+* `E` is a [*conditional expression*](https://github.com/dotnet/csharpstandard/blob/standard-v6/standard/expressions.md#1115-conditional-operator) `b ? x : y` where one or both of `x` and `y` is a *collection expression* and for each of the operands `x` and `y` one of the following hold:
+  * The operand is a collection expression and for each element `Eᵢ` there is an implicit conversion to `T`.
+  * The operand has an [*iteration type*](https://github.com/dotnet/csharpstandard/blob/standard-v6/standard/statements.md#1295-the-foreach-statement) `Tₑ` and there is an implicit conversion from `Tₑ` to `T`.
```

Comment:
so, one thing i don't like about this formalization is:

1. it special cases condtional-expressions, but not something like switch-expressions.
2. it's non-recursive.  If you had:  `byte[] y = [.. a ? b ? [1] : [2] : [3]];`, then i would expect that to work as well.

Basically, my preference would be that instead of special casing *only* the exact pattern of `a ? b : c`, we write such that this is more generalized.